File: web-admin/src/services/diseaseService.js
import api from './api';

export const diseaseService = {
  getAll: (page = 0, size = 10, search = '') =>
    api.get(`/diseases?page=${page}&size=${size}&search=${search}`),
  getById: (id) => api.get(`/diseases/${id}`),
  create: (data) => api.post('/diseases', data),
  update: (id, data) => api.put(`/diseases/${id}`, data),
  delete: (id) => api.delete(`/diseases/${id}`),
};

File: backend/src/main/java/vn/edu/fpt/doghandbook/backend/service/DiseaseService.java
package vn.edu.fpt.doghandbook.backend.service;

import vn.edu.fpt.doghandbook.backend.dto.request.DiseaseRequest;
import vn.edu.fpt.doghandbook.backend.dto.response.DiseaseResponse;
import vn.edu.fpt.doghandbook.backend.dto.response.PageResponse;

public interface DiseaseService {

    PageResponse<DiseaseResponse> getAll(int page, int size, String search);

    DiseaseResponse getById(Integer id);

    DiseaseResponse create(DiseaseRequest request, Integer createdByUserId);

    DiseaseResponse update(Integer id, DiseaseRequest request);

    void delete(Integer id);
}
